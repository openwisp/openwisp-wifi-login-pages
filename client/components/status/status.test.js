/* eslint-disable prefer-promise-reject-errors */
import axios from "axios";
/* eslint-disable camelcase */
import {shallow} from "enzyme";
import React from "react";
import {Cookies} from "react-cookie";
import ShallowRenderer from "react-test-renderer/shallow";

import getConfig from "../../utils/get-config";
import Status from "./status";

jest.mock("axios");

const defaultConfig = getConfig("default");
const createTestProps = props => {
  return {
    language: "en",
    orgSlug: "default",
    statusPage: defaultConfig.components.status_page,
    cookies: new Cookies(),
    logout: jest.fn(),
    ...props,
  };
};

describe("<Status /> rendering", () => {
  let props;
  it("should render correctly", () => {
    props = createTestProps();
    const renderer = new ShallowRenderer();
    const component = renderer.render(<Status {...props} />);
    expect(component).toMatchSnapshot();
  });
});

describe("<Status /> interactions", () => {
  let props;
  let wrapper;
  let originalError;
  let lastConsoleOutuput;
  beforeEach(() => {
    originalError = console.error;
    lastConsoleOutuput = null;
    console.error = (data) => {
      lastConsoleOutuput = data;
    };
  });
  afterEach(() => {
    console.error = originalError;
  });
  it("should call logout function when logout button is clicked", () => {
    axios.mockImplementationOnce(() => {
      return Promise.resolve({
        status: 200,
        statusText: 'OK',
        data: {
          "control:Auth-Type": "Accept",
        }
      });
    });
    props = createTestProps();
    wrapper = shallow(<Status {...props} />);
    wrapper.find("#owisp-status-logout-btn").simulate("click", {});
    expect(wrapper.instance().props.logout.mock.calls.length).toBe(1);
  });
  it("test componentDidMount lifecycle method", () => {
    axios
      .mockImplementationOnce(() => {
        return Promise.reject({
          status: 500,
          statusText: 'Internal Server Error',
          response: {
            data: {}
          }
        });
      })
      .mockImplementationOnce(() => {
        return Promise.reject({
          status: 504,
          statusText: "Gateway Timeout",
          response: {
            data: {}
          },
        });
      })
      .mockImplementationOnce(() => {
        return Promise.resolve({
          data: {
            "control:Auth-Type": "Accept",
          },
        });
      })
      .mockImplementationOnce(() => {
        return Promise.resolve({
          response: {
            status: 200,
            statusText: 'OK',
          },
          data: {
            "control:Auth-Type": "Reject",
          },
        });
      });
    props = createTestProps();
    wrapper = shallow(<Status {...props} />);
    return wrapper
      .instance()
      .componentDidMount()
      .then(() => {
        expect(wrapper.instance().props.logout.mock.calls.length).toBe(2);
        expect(lastConsoleOutuput).not.toBe(null);
        lastConsoleOutuput = null;
        return wrapper
          .instance()
          .componentDidMount()
          .then(() => {
            expect(wrapper.instance().props.logout.mock.calls.length).toBe(2);
            expect(lastConsoleOutuput).toBe(null);
            lastConsoleOutuput = null;
            return wrapper
              .instance()
              .componentDidMount()
              .then(() => {
                expect(wrapper.instance().props.logout.mock.calls.length).toBe(
                  3,
                );
                expect(lastConsoleOutuput).not.toBe(null);
                lastConsoleOutuput = null;
              });
          });
      });
  });
});
