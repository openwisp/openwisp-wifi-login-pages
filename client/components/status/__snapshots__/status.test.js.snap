// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Status /> rendering should render correctly 1`] = `
<React.Fragment>
  <InfoModal
    active={false}
    content={
      <p
        className="message"
      >
        Do you want to automatically log in the next time?
      </p>
    }
    handleResponse={[Function]}
    toggleModal={[Function]}
  />
  <div
    className="container content"
    id="status"
  >
    <div
      className="inner"
    >
      <div
        className="main-column"
      >
        <div
          className="inner"
        >
          <p
            className="status-content"
          >
            WiFi Login Successful!
          </p>
          <p
            className="status-content"
          >
            You can now use the internet.
          </p>
          <p
            className="status-content"
          >
            You may leave this page open in case you want to log out.
          </p>
          <p>
            <label>
              Status
              :
            </label>
            <span>
              Logged in
            </span>
          </p>
          <div
            className="row logout"
          >
            <input
              className="button full"
              onClick={[Function]}
              type="button"
              value="Logout"
            />
          </div>
          <div
            className="links row"
          >
            <Link
              className="button full status-link"
              to="/default/change-password/"
            >
              Change your password
            </Link>
          </div>
        </div>
      </div>
      <Memo(Connect(Contact)) />
    </div>
  </div>
  <div
    className="flex-column"
    id="sessions"
  >
    <loader
      full={false}
      small={true}
    />
  </div>
  <React.Fragment>
    <form
      action="http://localhost/"
      className="hidden"
      id="cp-login-form"
      method="post"
      target="owisp-auth-iframe"
    >
      <input
        name="auth_user"
        readOnly={true}
        type="text"
        value=""
      />
      <input
        name="auth_pass"
        readOnly={true}
        type="text"
        value=""
      />
      <input
        name="zone"
        readOnly={true}
        type="text"
        value="zone_name"
      />
      <input
        name="redirurl"
        readOnly={true}
        type="text"
        value="http://localhost:8080/default/status"
      />
      <input
        name="accept"
        readOnly={true}
        type="text"
        value="accept"
      />
    </form>
    
  </React.Fragment>
  <React.Fragment>
    <form
      action="http://localhost:8081"
      className="hidden"
      id="cp-logout-form"
      method="post"
      target="owisp-auth-logout-iframe"
    >
      <input
        name="logout_id"
        readOnly={true}
        type="hidden"
        value=""
      />
    </form>
    <iframe
      className="hidden"
      name="owisp-auth-logout-iframe"
      onLoad={[Function]}
      title="owisp-auth-iframe"
    />
  </React.Fragment>
</React.Fragment>
`;

exports[`<Status /> rendering with placeholder translation tags should render translation placeholder correctly 1`] = `
<React.Fragment>
  <InfoModal
    active={false}
    content={
      <p
        className="message"
      >
        LOGOUT_MODAL_CONTENT
      </p>
    }
    handleResponse={[Function]}
    toggleModal={[Function]}
  />
  <div
    className="container content"
    id="status"
  >
    <div
      className="inner"
    >
      <div
        className="main-column"
      >
        <div
          className="inner"
        >
          <p
            className="status-content"
          >
            STATUS_CONTENT
          </p>
          <p>
            <label>
              STATUS
              :
            </label>
            <span>
              LOGGED_IN
            </span>
          </p>
          <div
            className="row logout"
          >
            <input
              className="button full"
              onClick={[Function]}
              type="button"
              value="LOGOUT"
            />
          </div>
          <div
            className="links row"
          >
            <Link
              className="button full status-link"
              to="/default/change-password/"
            >
              Change your password
            </Link>
          </div>
        </div>
      </div>
      <Memo(Connect(Contact)) />
    </div>
  </div>
  <div
    className="flex-column"
    id="sessions"
  >
    <loader
      full={false}
      small={true}
    />
  </div>
  <React.Fragment>
    <form
      action="http://localhost/"
      className="hidden"
      id="cp-login-form"
      method="post"
      target="owisp-auth-iframe"
    >
      <input
        name="auth_user"
        readOnly={true}
        type="text"
        value=""
      />
      <input
        name="auth_pass"
        readOnly={true}
        type="text"
        value=""
      />
      <input
        name="zone"
        readOnly={true}
        type="text"
        value="zone_name"
      />
      <input
        name="redirurl"
        readOnly={true}
        type="text"
        value="http://localhost:8080/default/status"
      />
      <input
        name="accept"
        readOnly={true}
        type="text"
        value="accept"
      />
    </form>
    
  </React.Fragment>
  <React.Fragment>
    <form
      action="http://localhost:8081"
      className="hidden"
      id="cp-logout-form"
      method="post"
      target="owisp-auth-logout-iframe"
    >
      <input
        name="logout_id"
        readOnly={true}
        type="hidden"
        value=""
      />
    </form>
    <iframe
      className="hidden"
      name="owisp-auth-logout-iframe"
      onLoad={[Function]}
      title="owisp-auth-iframe"
    />
  </React.Fragment>
</React.Fragment>
`;
